### PR TITLE
Fix potential NULL pointer dereference before calling xmlXPathEvalExp…

### PR DESCRIPTION
**Pull request by icy17**
…ression

### CODE REVIEW

File: src/DS/sds.c [hunk 1]
```diff
@@ -791,7 +791,14 @@ static int ds_sds_compose_catalog_has_uri(xmlDocPtr doc, xmlNodePtr catalog, con
 	xpathCtx->node = catalog;
 
 	char *expression = oscap_sprintf("cat:uri[@uri = '%s']", uri);
+	if (expression == NULL)
+	{
+		oscap_seterr(OSCAP_EFAMILY_XML, "Error: Unable to create XPath expression.");
```
Comment (evgenyz) [inline src/DS/sds.c:796]: Possible problem's origin is not `libxml` or XML-related code. I believe error's affinity should be `OSCAP_EFAMILY_OSCAP`.

File: src/DS/sds.c [hunk 2]
```diff
@@ -1012,6 +1019,13 @@ static int ds_sds_compose_has_component_ref(xmlDocPtr doc, xmlNodePtr datastream
 	xpathCtx->node = datastream;
 
 	char *expression = oscap_sprintf("*/ds:component-ref[@xlink:href = '#%s' and @id = '%s']", filepath, cref_id);
+	if (expression == NULL)
+	{
+		oscap_seterr(OSCAP_EFAMILY_XML, "Error: Unable to create XPath expression.");
```
Comment (evgenyz) [inline src/DS/sds.c:1024]: Ditto.

**Review by evgenyz — CHANGES_REQUESTED** [review]
LGTM. Thank you!

File: src/DS/sds.c (hunk 1 above)
Comment (icy17) [reply src/DS/sds.c]: I've changed `OSCAP_EFAMILY_XML` to `OSCAP_EFAMILY_OSCAP`

**Comment by icy17** [comment]
Hi! Thanks for merging my pull requests!
I noticed these PR addresse potential security issues. I'm wondering if it would be appropriate to apply for CVEs for these bugs? Could you let me know if that's possible and what steps I should take?

**Pull request merged**